### DISCUSSION TITLE
Xbox.ts shows instock all the time

### DISCUSSION THREAD

**Discussion by Jwoods2019** [jef/streetmerchant#2475] [Help] [answered]
Hello I need a little help I'm trying to scrape the xbox store and it constantly shows instock does anyone have a xbox.ts working script that would mind sharing with me or helping me with Jeff's that would be greatly appreciated thanks.

**Comment by Jwoods2019** [comment]
Just checking back if anybody had any solution for this thanks in advance.

**Comment by Aragorn450** [comment, marked as answer]
So it looks like the Xbox store changed their layout and the way it checks if it is in stock or not is using a selector for the Out of Stock button. But that selector doesn't work any more with the updated layout and so that's why it is showing as in stock; The Out of Stock isn't found and therefore it's assumed it's In Stock.

To fix it, you can update `src/store/model/xbox.ts` and change the text in quotes in the `container` section to the following:

`'._-_-node_modules--xbox-web-partner-core-build-pages-BundleBuilder-Components-BundleBuilderHeader-__BundleBuilderHeader-module___checkoutButton'`

Make sure the comma is still there after that part. So the whole file should look like this:

```javascript
import {Store} from './store';

export const Xbox: Store = {
  currency: '$',
  labels: {
    outOfStock: {
      container:
        '._-_-node_modules--xbox-web-partner-core-build-pages-BundleBuilder-Components-BundleBuilderHeader-__BundleBuilderHeader-module___checkoutButton',
      text: ['out of stock'],
    },
  },
  links: [
    {
      brand: 'microsoft',
      model: 'xbox series x',
      series: 'xboxsx',
      url: 'https://www.xbox.com/en-us/configure/8WJ714N3RBTL',
    },
  ],
  name: 'xbox',
};

```

I notice that this isn't tracking the Xbox Series S, but perhaps that's because it's easily found? If you DO want it to track the S, the whole file should be this instead:

```javascript
import {Store} from './store';

export const Xbox: Store = {
  currency: '$',
  labels: {
    outOfStock: {
      container:
        '._-_-node_modules--xbox-web-partner-core-build-pages-BundleBuilder-Components-BundleBuilderHeader-__BundleBuilderHeader-module___checkoutButton',
      text: ['out of stock'],
    },
  },
  links: [
    {
      brand: 'microsoft',
      model: 'xbox series x',
      series: 'xboxsx',
      url: 'https://www.xbox.com/en-us/configure/8WJ714N3RBTL',
    },
    {
      brand: 'microsoft',
      model: 'xbox series s',
      series: 'xboxss',
      url: 'https://www.xbox.com/en-us/configure/942J774TP9JN',
    },
  ],
  name: 'xbox',
};

```

Although, because the S is in stock, you'll always get alerts about it as well. Anyway, hope that all helps you out!

**Reply by Jwoods2019** [reply]
Thank you so much i will give it shot!

**Reply by Aragorn450** [reply]
So did this work for you? If so, would you mind marking it as the answer so this isn't still marked as "unanswered"?

If it DIDN'T work for you, let me know and I'll do what I can to further help out.

**Reply by Jwoods2019** [reply]
Yes its working like a charm thanks again.

**Reply by Jwoods2019** [reply]
looks like something has change would you mind taking a look for me thanks.

**Reply by The-Synthax** [reply]
> looks like something has change would you mind taking a look for me thanks.

        '[class="BundleBuilderHeader-module__checkoutButton___3UyEq w-100 bg-light-green btn btn-primary"]',
      text: ['Out of stock'],

This seems to be working for me.
